class.jetpack-post-images.php

```diff
@@ -91,19 +91,38 @@ static function from_gallery( $post_id ) {
 
 		$permalink = get_permalink( $post->ID );
 
-		$gallery_images = get_post_galleries_images( $post->ID, false );
-
-		foreach ( $gallery_images as $galleries ) {
-			foreach ( $galleries as $src ) {
-				list( $raw_src ) = explode( '?', $src ); // pull off any Query string (?w=250)
-				$raw_src = wp_specialchars_decode( $raw_src ); // rawify it
-				$raw_src = esc_url_raw( $raw_src ); // clean it
-				$images[] = array(
-					'type'  => 'image',
-					'from'  => 'gallery',
-					'src'   => $raw_src,
-					'href'  => $permalink,
-				);
+		$galleries = get_post_galleries( $post->ID, false );
+
+		foreach ( $galleries as $gallery ) {
+			if ( isset( $gallery['type'] ) && 'slideshow' === $gallery['type'] && ! empty( $gallery['ids'] ) ) {
+				$image_ids = explode( ',', $gallery['ids'] );
+				$image_size = isset( $gallery['size'] ) ? $gallery['size'] : 'thumbnail';
+				foreach ( $image_ids as $image_id ) {
+					$image = wp_get_attachment_image_src( $image_id, $image_size );
+					if ( ! empty( $image[0] ) ) {
+						list( $raw_src ) = explode( '?', $image[0] ); // pull off any Query string (?w=250)
+						$raw_src = wp_specialchars_decode( $raw_src ); // rawify it
+						$raw_src = esc_url_raw( $raw_src ); // clean it
+						$images[] = array(
+							'type'  => 'image',
+							'from'  => 'gallery',
```

Comment:
Good observation. I'd leave it as `gallery` since it's extracted from a gallery, which happens to be of `slideshow` type, but it's first and foremost a gallery. I'd also like to avoid changing the output as little as possible in case other code using this is expecting a `from` of `gallery` type.